source/daplink/cmsis-dap/DAP.c

```diff
@@ -168,6 +191,8 @@ static uint32_t TimerTick;
 
 // Start Timer
 static __inline void TIMER_START (uint32_t usec) {
+	if (!DAP_Lock_Verify(0)) return 0;
```

Comment:
This is application specific code that probably shouldn't be in this file. If you want to add checking I would leave it as a sanity check assert which doesn't effect program flow. This shouldn't be hit in practice if locking is done correctly in the layer above.